references/segmentation/train.py

```diff
@@ -157,8 +153,15 @@ def main(args):
         optimizer,
         lambda x: (1 - x / (len(data_loader) * args.epochs)) ** 0.9)
 
+    if args.resume:
+        checkpoint = torch.load(args.resume, map_location='cpu')
+        model_without_ddp.load_state_dict(checkpoint['model'])
```

Comment:
@fmassa -- I haven't tested that moving the moment that `load_state_dict` will be loaded still works with DDP. Do you want to comment on this?

Reply:
Great question.

TL;DR: yes, this works with DDP in the current setup, and is what we do for the classification and detection reference scripts as well, see https://github.com/pytorch/vision/blob/1affa2e8d8ffdf4f6e340f9f29de2ff06b937db7/references/classification/train.py#L191-L196 and https://github.com/pytorch/vision/blob/1affa2e8d8ffdf4f6e340f9f29de2ff06b937db7/references/detection/train.py#L112-L117

The reason why this works for DDP is that we only use 1 GPU per python process. In this case, the same code gets executed, meaning that all the GPUs will load the checkpoint and get updated accordingly. This wouldn't be the case if we were using 2 GPUs per python process (à la `DataParallel` for example), so this is important to keep in mind.